### PR TITLE
RFR: Enhancement in nox

### PR DESCRIPTION
**Pull request by digitronik**
- Restore cli
- Parametrize nox over python versions
- Added pre-commit run with nox
- Update README

### CODE REVIEW

**Comment by digitronik** [comment]
@akshay196 take look.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -43,8 +43,10 @@ jobs:
           python-version: ${{ matrix.python-version }}
           architecture: x64
 
-      - name: Install nox
-        run: python -m pip install nox
+      - name: Setup Package and Install Devel Dependancies
```
Comment (akshay196) [inline .github/workflows/main.yml]: Fix spelling to `Dependencies`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -51,15 +51,22 @@ When contributing to linkstatus, we ask that you:
...
 source .venv/bin/activate
-pip install -e .
+python -m pip install -Ur dev-requirements.txt
+python -m pip install -e .
 linkstatus --help
 ```
 
+# Run unit test with nox
+```shell
+nox --list          # list testing env
```
Comment (akshay196) [inline CONTRIBUTING.md]: [optional] More appropriate comment here is `# List all available sessions`

**Comment by akshay196** [comment]
@digitronik  Hey. I am not user of pre-commit. Just a question pop up in my mind: what is purpose of providing pre-commit check in nox session? How it is useful for a developer as s/he can directly execute commands `pre-commit run -a` in his/her shell?

**Comment by digitronik** [comment]
> @digitronik Hey. I am not user of pre-commit. Just a question pop up in my mind: what is purpose of providing pre-commit check in nox session? How it is useful for a developer as s/he can directly execute commands `pre-commit run -a` in his/her shell?

Actually, `nox` session indirectly run by any developer independently. `nox` just help to s/he for collective operations with respective python env.

if someone run `nox` only; It will run all sessions one by one including `pre-commit`.

I can say completely optional.

**Pull request merged**